publish.gradle

```
@@ -0,0 +1,11 @@
+apply plugin: "maven-publish"
+apply plugin: "binary-compatibility-validator"
+
+publishing {
+	repositories {
+		maven {
+			name = "TestMaven"
+			url = file("${layout.buildDirectory}/testMaven").toURI().toString()
```

Changed deprecated `buildDir` to an alternative `layout.buildDirectory`

---

Now that it executes in the context of the subproject, we need to use the root projects' build dir. Also skipped a roundtrip through string with some of the `Directory` APIs.
```suggestion
			url = rootProject.layout.buildDirectory.dir("testMaven").get().asFile.toURI().toString()
```